### PR TITLE
[doc] Implementation details proposal

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

File: docs/implementation.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Implementation
+
+## Goal
+
+1. Online event-based diff test
+2. Decoupled
```
Comment (sequencer) [inline docs/implementation.md]: Decouple what?

File: docs/implementation.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+3. Serialization
```
Comment (sequencer) [inline docs/implementation.md]: Serialize what

File: docs/implementation.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
```
Comment (sequencer) [inline docs/implementation.md:22]: You can add an overview of different CPU tests infrastructural
1. chipsalliance/rocket-chip like Trace IO
2. sequencer/vector like online difftest with grand central DPI
3. FireSim
4. traditional SystemVerilog testbench

Comment (ksco) [reply docs/implementation.md:22]: Maybe just an overview of different co-simulation frameworks?

**Comment by ksco** [comment]
Closed as no progress has been made for too long.

**Pull request closed without merging**